dirty_cat/string_distances.py

```diff
@@ -19,6 +20,52 @@
 # TODO vectorize these functions (accept arrays)
 
 
+def get_ngram_count(X, ngram_range):
+    """Compute the number of ngrams in a string.
+
+    Here is where the formula comes from:
+
+    * the number of 3-grams in a string is the number of sliding windows of
+      size 3 in the string: len(string) - 3 + 1
+    * this can be generalized to n-grams by changing 3 by n.
+    * when given a ngram_range, we can sum this formula over all possible
+      ngrams
+
+    """
+    min_n, max_n = ngram_range
+    ngram_count = 0
+
+    for i in range(min_n, max_n + 1):
+        ngram_count += len(X) - i + 1
+
+    return ngram_count
+
+
+def preprocess(x):
```

Comment:
I think that we need a more explicit name. Maybe "count_vectorizer_preprocess".

Reply:
We can choose a more explicit name, but `count_vectorizer_preprocess` may be misleading as some of the preprocessing done was done by `ngram_similarity` (adding spaces at the extremities of the string)